libtenzir/builtins/operators/chart2.cpp

```diff
@@ -584,29 +590,30 @@ class chart_operator2 final : public crtp_operator<chart_operator2> {
         .hint(
           "consider filtering data or aggregating over a bigger `resolution`")
         .emit(ctx);
+      return nullptr;
     }
     return &map[xv];
   }
 
-  auto get_bucket(group_map& map, const plugins_map& plugins, const data_view x,
-                  const std::string_view group, session ctx) const -> bucket* {
+  auto get_bucket(group_map& map, const data_view& x,
+                  const std::string_view group, session ctx) const
+    -> std::pair<bucket*, bool> {
```

Comment:
This is hard to understand. Given that you never return `{non_null_ptr, true}`, I assume this could be something like `std::optional<bucket*>`? Or maybe it would it make sense if this function directly created the new bucket? Then you can just make it return `bucket*` and the other code also becomes easier to understand.